x-pack/plugins/security_solution/public/common/components/markdown_editor/plugins/insight/index.tsx

```diff
@@ -117,12 +118,23 @@ const InsightComponent = ({ label, description, providers }: InsightComponentPro
   const { totalCount, isQueryLoading, oldestTimestamp, hasError } = useInsightQuery({
     dataProviders,
   });
-  const timerange: AbsoluteTimeRange = useMemo(() => {
-    return {
-      kind: 'absolute',
-      from: oldestTimestamp ?? '',
-      to: new Date().toISOString(),
-    };
+  const timerange: TimeRange = useMemo(() => {
+    if (oldestTimestamp != null) {
+      return {
+        kind: 'absolute',
+        from: oldestTimestamp,
+        to: new Date().toISOString(),
+      };
+    } else {
+      const { to, from, fromStr, toStr } = getTimeRangeSettings();
```

Comment:
Overall changes look good. But are the timeranges returned from `getTimeRangeSettings` always relative? If it is so, should we consider adding `kind` as part of object returned from `getTimeRangeSettings`

Reply:
appears that they don't have to be, not sure how well this would work however...
<img width="953" alt="image" src="https://user-images.githubusercontent.com/56408403/204576376-4ab94f11-963e-41c5-b9b0-99bbeb4b67ee.png">